scripts/qa_crowbarsetup.sh

```diff
@@ -4208,6 +4208,58 @@ EOF
     exit $ret
 }
 
+function ping_fips
+{
+    local fips=$(openstack ip floating list -f value -c IP)
+    for fip in $fips; do
+        ping -c 1 -w 60 $fip || complain 120 "cannot reach test VM at $fip."
+    done
+}
+
+function oncontroller_testpreupgrade
+{
+    heat stack-create upgrade_test -f /root/scripts/heat/main.yaml -P "num_networks=2;num_instances=1"
+    wait_for 15 20 "heat stack-list | grep upgrade_test | grep CREATE_COMPLETE" \
+             "waiting for heat stack for upgrade tests to complete"
```

Comment:
`wait_for` already prefixes the messages with "waiting for:" to remove duplication